src/Avalonia.Base/Input/GestureRecognizers/HoldGestureRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+        private PointerPressedEventArgs? _pointerEventArgs;
+        private Rect _trackingBounds;
+        private Timer? _holdTimer;
+        private bool _elasped;
+
+        /// <summary>
+        /// Defines the <see cref="IsHoldWithMouseEnabled"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsHoldWithMouseEnabledProperty =
+            AvaloniaProperty.Register<HoldGestureRecognizer, bool>(
+                nameof(IsHoldWithMouseEnabled));
+        
+        /// <summary>
+        /// Gets or sets whether to detect hold from the mouse
+        /// </summary>
+        public bool IsHoldWithMouseEnabled
+        {
+            get => GetValue(IsHoldWithMouseEnabledProperty);
+            set => SetValue(IsHoldWithMouseEnabledProperty, value);
+        }
+
+        public void Initialize(IInputElement target, IGestureRecognizerActionsDispatcher actions)
+        {
+            _target = target;
+            _actions = actions;
+
+            _target?.AddHandler(InputElement.PointerPressedEvent, OnPointerPressed, Interactivity.RoutingStrategies.Tunnel | Interactivity.RoutingStrategies.Bubble);
+            _target?.AddHandler(InputElement.PointerReleasedEvent, OnPointerReleased, Interactivity.RoutingStrategies.Tunnel | Interactivity.RoutingStrategies.Bubble);
+
+            _holdTimer = new Timer(300);
+            _holdTimer.AutoReset = false;
+            _holdTimer.Elapsed += HoldTimer_Elapsed;
+        }
+
+        private async void HoldTimer_Elapsed(object? sender, ElapsedEventArgs e)
+        {
+            _elasped = true;
+            _holdTimer?.Stop();
+
+            if(_tracking != null)
+            {
+                await Dispatcher.UIThread.InvokeAsync(() => _target?.RaiseEvent(new HoldGestureEventArgs(_gestureId, _pointerEventArgs, HoldingState.Started)));
+            }
+        }
+
+        private void OnPointerPressed(object? sender, PointerPressedEventArgs e)
+        {
+            PointerPressed(e);
+        }
+
+        private void OnPointerReleased(object? sender, PointerReleasedEventArgs e)
+        {
+            PointerReleased(e);
+        }
+
+        public void PointerCaptureLost(IPointer pointer)
+        {
+            if (_tracking == pointer)
+            {
+                EndHold(!_elasped);
+            }
+        }
+
+        public void PointerMoved(PointerEventArgs e)
+        {
+            if (_tracking == e.Pointer && _target is Visual visual)
+            {
+                var currentPosition = e.GetPosition(visual);
+
+                if (!_trackingBounds.Contains(currentPosition))
+                {
+                    EndHold(true);
+                }
+            }
+        }
+
+        public void PointerPressed(PointerPressedEventArgs e)
+        {
+            if (_target != null && _target is Visual visual && (IsHoldWithMouseEnabled || e.Pointer.Type == PointerType.Touch || e.Pointer.Type == PointerType.Pen))
```

Comment:
Let's check for the pointer type first, so property read won't be called on touch/pen devices (for the minor perf improvement). As avalonia property reads are not free.